src/main/java/Duke.java

```diff
@@ -1,10 +1,224 @@
+                saveTasks(list);
+            } else if (userInput.startsWith("todo")) {
+                try {
+                    if (userInput.length() < 6) {
+                        throw new DukeException("Remember to add an actual task. Try again!");
+                    }
+                } catch (DukeException e) {
+                    System.err.println(e.getMessage());
+                    continue;
+                }
+                System.out.println(line);
+                String todoTask = userInput.substring(5);
+                ToDos newTodo = new ToDos(todoTask);
+                list.add(newTodo);
+                newTodo.addTask(list.size());
+                System.out.println(line);
+                saveTasks(list);
+            } else if (userInput.startsWith("deadline")) {
+                try {
+                    // other improvements: no content, empty deadline
+                    if (userInput.length() < 10) {
+                        throw new DukeException("Remember to add an actual task. Try again!");
+                    }
+                    if (!(userInput.contains("/by"))) {
+                        throw new DukeException("Remember to state the deadline after a '/by'!");
+                    }
+                } catch (DukeException e) {
+                    System.err.println(e.getMessage());
+                    continue;
+                }
+                System.out.println(line);
+                int preIdx = userInput.indexOf("/by");
+                int idx = preIdx + 4;
+                String deadlineTask = userInput.substring(9, preIdx - 1);
+                String deadlineBy = userInput.substring(idx);
+                LocalDate deadlineDate = LocalDate.parse(deadlineBy);
+                Deadline newDeadline = new Deadline(deadlineTask, deadlineDate);
+                list.add(newDeadline);
+                newDeadline.addTask(list.size());
+                System.out.println(line);
+                saveTasks(list);
+            } else if (userInput.startsWith("event")) {
+                try {
+                    // other improvements: no content, empty start and end
+                    if (userInput.length() < 7) {
+                        throw new DukeException("Remember to add an actual task. Try again!");
+                    }
+                    if (!(userInput.contains("/from"))) {
+                        throw new DukeException("Remember to state the start of your event after a '/from'!");
+                    }
+                    if (!(userInput.contains("/to"))) {
+                        throw new DukeException("Remember to state the end of your event after a '/to'!");
+                    }
+                } catch (DukeException e) {
+                    System.err.println(e.getMessage());
+                    continue;
+                }
+                System.out.println(line);
+                int preIdxFrom = userInput.indexOf("/from");
```

Comment:
Perhaps some of the upcoming lines can be separated by one blank line, such as between variable definitions and calling methods.
e.g. 
```java
                LocalDate eventDateTo = LocalDate.parse(eventTo);
                Event newEvent = new Event(eventTask, eventDateFrom, eventDateTo);

                list.add(newEvent);
                newEvent.addTask(list.size());

                System.out.println(line);
```